lib/exfmt.ex

```diff
@@ -83,9 +83,20 @@ defmodule Exfmt do
   """
   @spec unsafe_format(String.t, integer) :: {:ok, String.t} | SyntaxError.t
   def unsafe_format(source, max_width \\ @max_width) do
+    leading_indent = leading_indent(source)
+    leading_whitespace = leading_whitespace(source, leading_indent)
+    trailing_whitespace = trailing_whitespace(source)
+
     with {:ok, tree} <- Code.string_to_quoted(source),
          {:ok, comments} <- Comment.extract_comments(source) do
-      {:ok, do_format(tree, comments, max_width)}
+      result =
+        tree
+        |> do_format(comments, max_width - leading_indent)
+        |> add_leading_indent(leading_indent)
+        |> add_trailing_whitespace(trailing_whitespace)
+        |> add_leading_whitespace(leading_whitespace)
+        |> strip_trailing_line_whitespace
```

Comment:
Few thoughts:

We could avoid splitting the output and concatenating whitespace onto each line by making the starting indent configurable in the printer.

When concatenating binaries together it's often faster to build up IO lists and then use `IO.iodata_to_binary/1` or `IO.chardat.to_string/1`.